tests/integration/test_segy_import_export.py

```diff
@@ -12,17 +12,77 @@
 from mdio import mdio_to_segy
 from mdio.converters import segy_to_mdio
 from mdio.core import Dimension
+from mdio.segy.geometry import StreamerShotGeometryType
 
 
 dask.config.set(scheduler="synchronous")
 
 
+@pytest.mark.parametrize("header_locations", [(17, 137)])
+@pytest.mark.parametrize("header_names", [("shot_point", "cable")])
+@pytest.mark.parametrize("header_types", [("int32", "int16")])
+@pytest.mark.parametrize("endian", ["big"])
+@pytest.mark.parametrize(
+    "grid_overrides", [{"NonBinned": True, "chunksize": 2}, {"HasDuplicates": True}]
+)
+@pytest.mark.parametrize("chan_header_type", [StreamerShotGeometryType.C])
+class TestImport4DNonReg:
+    """Test for 4D segy import with grid overrides."""
+
+    def test_import_4d_segy(
+        self,
+        segy_mock_4d_shots,
+        zarr_tmp,
+        header_locations,
+        header_names,
+        header_types,
+        endian,
+        grid_overrides,
+        chan_header_type,
+    ):
+        """Test importing a SEG-Y file to MDIO."""
+        segy_path = segy_mock_4d_shots[chan_header_type]
+
+        segy_to_mdio(
+            segy_path=segy_path,
+            mdio_path_or_buffer=zarr_tmp.__str__(),
+            index_bytes=header_locations,
+            index_names=header_names,
+            index_types=header_types,
+            chunksize=(8, 2, 10),
+            overwrite=True,
+            endian=endian,
+            grid_overrides=grid_overrides,
+        )
+
+        # Expected values
+        num_samples = 25
+        shots = [2, 3, 5]
+        cables = [0, 101, 201, 301]
+        receivers_per_cable = [1, 5, 7, 5]
+
+        # QC mdio output
+        mdio = MDIOReader(zarr_tmp.__str__(), access_pattern="0123")
+        assert mdio.binary_header["Samples"] == num_samples
+        grid = mdio.grid
+
+        assert grid.select_dim(header_names[0]) == Dimension(shots, header_names[0])
+        assert grid.select_dim(header_names[1]) == Dimension(cables, header_names[1])
+        assert grid.select_dim("trace") == Dimension(
+            range(1, np.amax(receivers_per_cable) + 1), "trace"
+        )
+        samples_exp = Dimension(range(0, num_samples, 1), "sample")
+        assert grid.select_dim("sample") == samples_exp
+
```

Comment:
I just noticed, on any of these, we don't verify if the sample values (amplitudes) and exported SEG-Ys are correct. Can we add them?

Amplitude checks would go to TestReader class in this file; and exports would go to TestExport class

Reply:
@markspec let's make a new PR for these